src/makeCalculator.js

```diff
@@ -4,7 +4,47 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    subtract(value) {
+      this.result -= value;
+
+      return this.result;
+    },
+
+    add(value) {
+      this.result += value;
+
+      return this.result;
+    },
+
+    multiply(value) {
+      this.result *= value;
+
+      return this.result;
+    },
+
+    divide(value) {
+      this.result /= value;
+
+      return this.result;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(callback, value) {
+      this.result = callback.call(this, value);
+
+      return this;
```

Comment:
The `operate` method correctly applies the callback to the `result` property and returns the calculator object itself for chaining. However, it is important to ensure that the callback is a function before calling it. This can be done by checking `typeof callback === 'function'` before executing the callback.